### PR TITLE
:bug: #390 Workaround to prevent SEGV when POM file has no properties

### LINKED ISSUE #390: [BUG] Property resolution leads to SEGV when POM has no own props
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Konveyor version

How can I get it? I am using Analyzer-LSP directly!

### Priority

Blocker

### Current Behavior

When analysing a Java POM with a dependency as a property, analyzer tries to resolve the property in the same POM. If the POM contains no properties at all, this leads to a segmentation violation (in [line 141 of `provider/internal/java/dependeny.go`](https://github.com/konveyor/analyzer-lsp/blob/9369083d47c8b056775300b38adeae856612f397/provider/internal/java/dependency.go#L141)):

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x5b3ddc]
```

### Expected Behavior

Try to resolve property via parent POM (hierarchy) or leave unresolved otherwise (which [is the default, if the property cannot be found in an existing POM property list](https://github.com/konveyor/analyzer-lsp/blob/9369083d47c8b056775300b38adeae856612f397/provider/internal/java/dependency.go#L142)).

### How Reproducible

Always (Default)

### Steps To Reproduce

Prepare a project with POM like this.

```xml
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>
    <groupId>some-project</groupId>
    <artifactId>some-artifact</artifactId>
    <version>does-not-matter</version>
    
    <dependencies>
        <dependency>
            <groupId>some-dep-group</groupId>
            <artifactId>some-dep-artifact</artifactId>
            <version>${cannot-be-resolved}</version>
        </dependency>
    </dependencies>
</project>
``` 

And analyze it.

### Environment

```markdown
Have to provide a minimal example ...
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by ascheman**
Fixes #390 
CAUTION: 
* This change only provides a workaround to avoid the SEGV of #390
* It does not find the property contents, the property search should be extended to the parent POM (hierarchy).

### CODE REVIEW

**Comment by ascheman** [comment]
I still would like to ask whether having two different dependency analysis approaches are the way to go (cf. #420).

File: provider/internal/java/dependency.go [hunk 1]
```diff
@@ -140,9 +143,20 @@ func (p *javaServiceClient) GetDependenciesFallback(ctx context.Context, locatio
...
+					"property", version,
+					"properties", pom.Properties)
+				if pom.Properties == nil {
+					p.log.Info("Cannot resolve version property value as POM does not have properties",
+						"POM", fmt.Sprintf("%s.%s", pomCoordinate(pom.GroupID), pomCoordinate(pom.ArtifactID)),
+						"property", version,
+						"dependency", dep.Name)
 					dep.Version = version
+				} else {
+					version = pom.Properties.Entries[version]
```
[the remark this answers is not in the document]
Comment (ascheman) [reply provider/internal/java/dependency.go:156]: I have no idea. Trying to provide proper test cases ran into other problems, cf. #427. 
Feel free to pick the examples provided there for further debugging into _empty_ properties.

[the remark this answers is not in the document]
Comment (pranavgaikwad) [reply provider/internal/java/dependency.go:156]: Looks like its never nil

**Review by pranavgaikwad — APPROVED** [review]
Thank you for the fix!

**Pull request merged**